src/index.js

```diff
@@ -157,6 +157,10 @@ class UglifyJsPlugin {
       uglifyOptions.output = uglifyOptions.output || {};
     }
 
+    if (typeof this.options.sourceMap === 'undefined' && ['sourcemap', 'source-map', 'hidden-source-map'].indexOf(compiler.options.devtool) > -1) {
```

Comment:
If `devtool` should be automatically supported, the previous `this.options.sourceMap`  'doesn't matter'  as it is overridden to `true` anyways in case `compilation.options.devtool` is set ? Maybe I'm missing a case here 🙃